app/components/ActivityBar.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/prefer-stateless-function */
+import React, { Component } from 'react';
+import {
+  StyleSheet, View,
+} from 'react-native';
+
+export default class ActivityBar extends Component {
+  render() {
+    let { activityLevel, barWidth } = this.props;
+
+    barWidth = parseInt(barWidth);
+
+    if (activityLevel === 'N/A') {
```

Comment:
You can do this logic before the view components, and then do a {var cond ? thing1 : thing2} This will clean up the view so that theres not so much logic in the middle of the components